spec/core/resource_spec.rb

```diff
@@ -455,7 +455,12 @@ def to_param
 
   context "#inspect" do
     it "should display nicely" do
-      expect(ZendeskAPI::User.new(client, :foo => :bar).inspect).to eq("#<ZendeskAPI::User {\"foo\"=>:bar}>")
+      expected_user_representation = if RUBY_VERSION >= "3.4"
+        "#<ZendeskAPI::User {\"foo\" => :bar}>"
+      else
+        "#<ZendeskAPI::User {\"foo\"=>:bar}>"
+      end
+      expect(ZendeskAPI::User.new(client, :foo => :bar).inspect).to eq(expected_user_representation)
```

Comment:
This can be done differently:
```ruby
expect(ZendeskAPI::User.new(client, :foo => :bar).inspect).to eq("#<ZendeskAPI::User #{{'foo' => :bar}}>")
```

That is, we interpolate a hash directly in the expected output, and it match Hash#inspect behaviour. However, here we only have one such test, and we explicitly are interested in how User#inspect behaves, so I thought it would be better to have two branches.